File: src/backgroundTasks/riot/FillSummoners.ts
import DBHelper from '../../helpers/DBHelper.js';
import RiotHelper from '../../helpers/RiotHelper.js';
import axiosRetry from 'axios-retry';
import axios from 'axios';
import { logger } from '../../config/logging.js';
import { SummonerDTO } from '../../interfaces/CustomInterfaces.js';

export default class FillSummonersTask {
  dbHelper: DBHelper;
  riotHelper: RiotHelper;
  accountNames: { name: string; tag: string; core: boolean }[];

  constructor(accountNames?: { name: string; tag: string; core: boolean }[]) {
    this.dbHelper = DBHelper.getInstance();
    this.riotHelper = RiotHelper.getInstance();
    axiosRetry(axios, {
      retries: 4, // number of retries
      retryDelay: (retryCount) => {
        logger.warn(`retry attempt: ${retryCount}`);
        return retryCount * 2000; // time interval between retries
      },
    });

    if (accountNames) {
      this.accountNames = accountNames;
    } else {
      this.accountNames = [
        {
          name: 'AngryBacteria',
          tag: 'cnap',
          core: true,
        },
        {
          name: 'BriBri',
          tag: '0699',
          core: true,
        },
        {
          name: 'VerniHD',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Baywack',
          tag: 'CnAP',
          core: true,
        },
        {
          name: '3 6 6 1',
          tag: '#EUW',
          core: false,
        },
        {
          name: 'SignisAura',
          tag: 'CnAP',
          core: true,
        },
        {
          name: 'Alraune22',
          tag: 'CnAP',
          core: true,
        },
        {
          name: 'Aw3s0m3mag1c',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Gnerfedurf',
          tag: 'BCH',
          core: true,
        },
        {
          name: 'Gnoblin',
          tag: 'BCH',
          core: true,
        },
        {
          name: 'VredVampire',
          tag: '2503',
          core: false,
        },
        {
          name: 'D3M0NK1LL3RG0D',
          tag: 'EUW',
          core: true,
        },
        {
          name: 'GLOMVE',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'hide on büschli',
          tag: 'EUW',
          core: true,
        },
        {
          name: 'IBlueSnow',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Nayan Stocker',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Norina Michel',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'pentaskill',
          tag: 'CnAP',
          core: true,
        },
        {
          name: 'Pollux',
          tag: '2910',
          core: true,
        },
        {
          name: 'Polylinux',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Prequ',
          tag: 'EUW',
          core: true,
        },
        {
          name: 'Sausage Revolver',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'swiss egirI',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'TCT Tawan',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'The 26th Bam',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Theera3rd',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'Zinsstro',
          tag: 'EUW',
          core: false,
        },
        {
          name: 'WhatThePlay',
          tag: 'CnAP',
          core: true,
        },
        {
          name: 'pentaskill',
          tag: 'CnAP',
          core: true,
        },
        {
          name: 'Árexo',
          tag: 'CNAP',
          core: true,
        },
      ];
    }
  }

  async fillSummoners() {
    const summonerObjects: SummonerDTO[] = [];
    for (const account of this.accountNames) {
      if (account.core) {
        const riotData = await this.riotHelper.getSummonerByAccountTag(
          account.name,
          account.tag,
        );
        if (riotData) {
          summonerObjects.push(riotData);
        }
      }
    }
    await this.dbHelper.updateSummoners(summonerObjects);
  }
}

const task = new FillSummonersTask();
void task.fillSummoners();

File: tasks/FillSummoners.py
import asyncio
from typing import List, Dict, Optional

from helpers.DBHelper import DBHelper
from helpers.RiotHelper import RiotHelper


class FillSummonersTask:
    def __init__(self):
        self.db_helper = DBHelper()
        self.riot_helper = RiotHelper()
        self.account_names = [
            {"name": "AngryBacteria", "tag": "cnap", "core": True},
            {"name": "BriBri", "tag": "0699", "core": True},
            {"name": "VerniHD", "tag": "EUW", "core": False},
            {"name": "Baywack", "tag": "CnAP", "core": True},
            {"name": "3 6 6 1", "tag": "#EUW", "core": False},
            {"name": "SignisAura", "tag": "CnAP", "core": True},
            {"name": "Alraune22", "tag": "CnAP", "core": True},
            {"name": "Aw3s0m3mag1c", "tag": "EUW", "core": False},
            {"name": "Gnerfedurf", "tag": "BCH", "core": True},
            {"name": "Gnoblin", "tag": "BCH", "core": True},
            {"name": "VredVampire", "tag": "2503", "core": False},
            {"name": "D3M0NK1LL3RG0D", "tag": "EUW", "core": True},
            {"name": "GLOMVE", "tag": "EUW", "core": False},
            {"name": "hide on büschli", "tag": "EUW", "core": True},
            {"name": "IBlueSnow", "tag": "EUW", "core": False},
            {"name": "Nayan Stocker", "tag": "EUW", "core": False},
            {"name": "Norina Michel", "tag": "EUW", "core": False},
            {"name": "pentaskill", "tag": "CnAP", "core": True},
            {"name": "Pollux", "tag": "2910", "core": True},
            {"name": "Polylinux", "tag": "EUW", "core": False},
            {"name": "Prequ", "tag": "EUW", "core": True},
            {"name": "Sausage Revolver", "tag": "EUW", "core": False},
            {"name": "swiss egirI", "tag": "EUW", "core": False},
            {"name": "TCT Tawan", "tag": "EUW", "core": False},
            {"name": "The 26th Bam", "tag": "EUW", "core": False},
            {"name": "Theera3rd", "tag": "EUW", "core": False},
            {"name": "Zinsstro", "tag": "EUW", "core": False},
            {"name": "WhatThePlay", "tag": "CnAP", "core": True},
            {"name": "pentaskill", "tag": "CnAP", "core": True},
            {"name": "Árexo", "tag": "CNAP", "core": True},
        ]

    async def fill_summoners(self, core_only=True):
        summoner_objects = []

        for account in self.account_names:
            if not account["core"] and core_only:
                continue
            summoner_data = await self.riot_helper.get_summoner_by_account_tag(
                account["name"], account["tag"]
            )
            if summoner_data is not None:
                summoner_objects.append(summoner_data)

        summoner_objects = [obj for obj in summoner_objects]
        await self.db_helper.update_summoners(summoner_objects)


async def main():
    task = FillSummonersTask()
    await task.fill_summoners()


if __name__ == "__main__":
    asyncio.run(main())
